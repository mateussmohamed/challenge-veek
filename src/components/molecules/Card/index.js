import React from 'react'

import {
  Icon,
  CardContainer,
  CardImage,
  CardContent,
  CardFooter,
  CardUser,
  CardUserImage,
  CardUserName,
  CardInformation,
  CardTimestamp,
} from 'atoms'

const Card = props => (
  <CardContainer {...props}>
    <CardImage />
    <CardContent>Lorem ipsum dolor</CardContent>
    <CardFooter>
      <CardUser>
        <CardUserImage />
        <CardUserName>BY JOHN SMITH</CardUserName>
      </CardUser>
      <CardInformation>
        <Icon name="clock" />
        <CardTimestamp>
          <p>25MIN</p>
          <p>AGO</p>
        </CardTimestamp>
      </CardInformation>
    </CardFooter>
  </CardContainer>
)

export default Card
