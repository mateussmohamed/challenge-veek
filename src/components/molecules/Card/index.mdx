---
name: Card
menu: Molecules
---

import { Playground, PropsTable } from 'docz'

import Card from './index'

# Card
<PropsTable of={Card} />

<Playground>
  <Card />
</Playground>
