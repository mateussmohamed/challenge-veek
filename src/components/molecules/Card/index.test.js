import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Card from './index'

describe('[<Card />]', () => {
  const wrap = (props = {}) => shallow(<Card {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap()

    expect(wrapper).toMatchSnapshot()
  })
})
