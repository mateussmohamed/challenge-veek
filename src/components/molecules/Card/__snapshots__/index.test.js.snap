// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Card />] matches to snapshot 1`] = `
<Card__CardContainer>
  <Card__Image />
  <Card__Content>
    Lorem ipsum dolor
  </Card__Content>
  <Card__Footer>
    <Card__User>
      <Card__UserImage />
      <Card__UserName>
        BY JOHN SMITH
      </Card__UserName>
    </Card__User>
    <Card__PostedIn>
      <Icon
        name="clock"
      />
      <Card__Timestamp>
        <p>
          25MIN
        </p>
        <p>
          AGO
        </p>
      </Card__Timestamp>
    </Card__PostedIn>
  </Card__Footer>
</Card__CardContainer>
`;
